tests/test_integration.py

```diff
@@ -17,6 +20,57 @@
     from http.server import BaseHTTPRequestHandler, HTTPServer
 
 
+class TestHandler(BaseHTTPRequestHandler):
+    num_requests = 0
+
+    requests = defaultdict(Queue)
+
+    @classmethod
+    def _add_request(cls, req):
```

Comment:
Is there any way to make this non-static?

Reply:
Possibly by returning something from self.setup_mock_server?

Reply:
I think this needs to stay a class method. Things are weird because the python standard library is weird: particularly `HTTPServer` takes in `handler` as a *class* instead of an instance, and so we don't actually control making instances of `TestServer`, and an *instance* of BaseHTTPRequestHandler actually represents a single incoming request. So the test case needs its own `.get_requests`, and `_add_request` scoped to the *subclass* that it defines, not to an instance.